src/service.rs

```diff
@@ -366,8 +474,46 @@ impl Service {
                                 }
                             }
                         }
+                        HandlerOut::RequestTimedOut(request_id, node_id) => {
```

Comment:
the logic of this match branch is long enough to be better moved to its own function